x/ccv/provider/keeper/relay.go

```diff
@@ -222,7 +222,17 @@ func (k Keeper) QueueVSCPackets(ctx sdk.Context) {
 
 	for _, chain := range k.GetAllConsumerChains(ctx) {
 		currentValidators := k.GetConsumerValSet(ctx, chain.ChainId)
-		nextValidators := k.ComputeNextEpochConsumerValSet(ctx, chain.ChainId, bondedValidators)
+
+		if topN, found := k.GetTopN(ctx, chain.ChainId); found {
```

Comment:
Can we de-duplicate this more? Lines 226 to 235 here are still the same as lines 293 to 301 in proposal.go (but it's already much better like this)

Reply:
Yeah, you are right, let's keep this in mind to maybe consolidate more in the future, but leave as is for now